File: product-review-api-command-go/internal/client/order_client.go
package client

import "example/product-review-api-command-go/internal/dto/response"

type OrderClient interface {
	ValidateOrder(customerID string, orderID string, productID string) response.OrderValidationResponse
}

type orderClient struct {
}

func NewOrderClient() OrderClient {
	return &orderClient{}
}

func (c *orderClient) ValidateOrder(customerID string, orderID string, productID string) response.OrderValidationResponse {
	valid := len(orderID) >= 4 && orderID[:4] == "ORD-" && orderID != "ORD-INVALID"
	status := "NOT_FOUND"
	if valid {
		status = "COMPLETED"
	}
	return response.OrderValidationResponse{
		Valid:       valid,
		OrderStatus: status,
	}
}

File: product-review-api/src/main/java/com/example/review/client/OrderClient.java
package com.example.review.client;

import com.example.review.dto.response.OrderValidationResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class OrderClient {

    private static final Logger log = LoggerFactory.getLogger(OrderClient.class);

    public OrderValidationResponse validateOrder(String customerId, String orderId, String productId) {
        log.info("Validating order ownership for customerId={}, orderId={}, productId={}", customerId, orderId, productId);

        boolean valid = orderId.startsWith("ORD-") && !orderId.equals("ORD-INVALID");
        return new OrderValidationResponse(valid, valid ? "COMPLETED" : "NOT_FOUND");
    }
}
